nodejs/README.adoc

```diff
@@ -0,0 +1,341 @@
+* https://nodejs.org/en/download/[Node.js 14.x^] is installed.
+
+[NOTE]
+====
+The example Node.js application in this quick start uses a https://kafka.js.org/[KafkaJS^] client by default. If you want to use a https://github.com/blizzard/node-rdkafka[node-rdkafka^] client, you must install Docker to run a specified container image and configure a development environment. To learn more, see the https://github.com/nodeshift-starters/reactive-example/blob/main/README.md[documentation] for the Node.js application.
+====
+
+// Condition out QS-only content so that it doesn't appear in docs.
+// All QS anchor IDs must be in this alternate anchor ID format `[#anchor-id]` because the ascii splitter relies on the other format `[id="anchor-id"]` to generate module files.
+ifdef::qs[]
+[#description]
+Learn how to use Node.js applications to produce and consume messages using a Kafka instance in {product-long}.
+
+[#introduction]
+Welcome to the quick start for {product-long} with Node.js. In this quick start, you'll learn how to use https://nodejs.org/en/about/[Node.js^] to produce messages to and consume messages from your Kafka instances in {product}.
+endif::[]
+
+
+[id="proc-importing-nodejs-sample-code_{context}"]
+== Importing the Node.js sample code
+
+For this quick start, you'll use sample code from the Nodeshift Application Starters https://github.com/nodeshift-starters/reactive-example[reactive-example^] repository in GitHub. After you understand the concepts and tasks in this quick start, you can use your own Node.js applications with {product} in the same way.
+
+.Procedure
+. On your computer, open a command-line window.
+. On the command line, clone the Nodeshift Application Starters https://github.com/nodeshift-starters/reactive-example[reactive-example^] repository from GitHub.
++
+.Cloning the reactive-example repository
+[source,subs="+attributes"]
+----
+$ git clone https://github.com/nodeshift-starters/reactive-example.git
+----
+. In your IDE, open the `reactive-example` directory corresponding to the repository that you cloned.
+
+ifdef::qs[]
+.Verification
+* Is the Node.js example application accessible in your IDE?
+endif::[]
+
+[id="proc-configuring-nodejs_{context}"]
+== Configuring the Node.js example application to connect to a Kafka instance
+
+To enable your Node.js application to access a Kafka instance, you need to configure a connection by specifying the following details:
+
+* The bootstrap server endpoint for your Kafka instance
+* The generated credentials for your {product} service account
+* The Simple Authentication and Security Layer (SASL) mechanism that the client will use to authenticate with the Kafka instance
+
+In this task, you'll create a new shell script called `set-environment.sh`. In this shell script, you'll set the required bootstrap server and client credentials as environment variables.
```

Comment:
```
In this task, you'll create a new file called `.env`. In this file you'll set the required bootstrap server and client credentials so that they can be used to connect to your Kafka instance.
```